app/assets/javascripts/components/search/result_group.js.coffee.erb

```diff
@@ -0,0 +1,67 @@
+{div, p, table, tr, td, br} = React.DOM
+
+window.SearchResultGroupClass = React.createClass
+  getInitialState: ->
+    jQuery.extend({loading: false}, @props)
+
+  onPaginationSelect: (page) ->
+    if page isnt @state.group.pagination.current_page
+      jQuery("##{@pageParam}").val(page)
+      query = jQuery('#material_search_form').serialize()
+
+      window.updateSearchUrl(query)
+
+      # strip out all material_types[] params
+      query = query.replace(/(^|&)material_types(\[|%5B)(\]|%5D)=[^&]*/, '')
+
+      # add back the one for this group's material type
+      query += "&material_types[]=#{@materialType}"
+
+      jQuery.ajax
+        dataType: "json"
+        url: Portal.API_V1.SEARCH
+        data: query
+        success: (response) => @updateState(response)
+
+      @setState({loading: true})
+
+  updateState: (groupData) ->
+    @replaceState({group: groupData.results[0]})
+
+  renderLoading: ->
+    (div {className: 'border_top'},
+      (p {}, 'Finding materials...')
+    )
+
+  renderResults: ->
+   group = @state.group
+   onSelect = (page) => @onPaginationSelect(page)
```

Comment:
I'm not sure this is needed, you could just put onSelect: @onPaginationSelect in the render body.